Update cloudinit script

rvanlaar opened the pull request:
This cloudinit script had become outdated.

Changed 4 things:
- docker repository had moved
- use current docker gpg key
- use current dokku gpg key
- move over to ubuntu focal LTS 20.20

**Note:** If this PR is just doc changes, please put [ci skip] in the body that way tests do not run.

jalcine commented:
Probably for sanity checking, do you have a `dokku report` of a built box on Dreamhost @rvanlaar?

rvanlaar commented:
Here's the report. One caveat. I used DigitalOcean not Dreamhost:

```$ doctl compute droplet create test-deploy --enable-monitoring  --image ubuntu-20-04-x64 --size s-1vcpu-1gb --region ams3 --user-data-file cloudinit.sh```

```
root@test-deploy:~# dokku report
-----> uname: Linux test-deploy 5.4.0-29-generic #33-Ubuntu SMP Wed Apr 29 14:32:27 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
-----> memory: 
                     total        used        free      shared  buff/cache   available
       Mem:            981         235         197           0         548         597
       Swap:             0           0           0
-----> docker version: 
       Client:
        Version:           19.03.8
        API version:       1.40
        Go version:        go1.13.8
        Git commit:        afacb8b7f0
        Built:             Wed Mar 11 23:42:35 2020
        OS/Arch:           linux/amd64
        Experimental:      false
       
       Server:
        Engine:
         Version:          19.03.8
         API version:      1.40 (minimum version 1.12)
         Go version:       go1.13.8
         Git commit:       afacb8b7f0
         Built:            Wed Mar 11 22:48:33 2020
         OS/Arch:          linux/amd64
         Experimental:     false
        containerd:
         Version:          1.3.3-0ubuntu2
         GitCommit:        
        runc:
         Version:          spec: 1.0.1-dev
         GitCommit:        
        docker-init:
         Version:          0.18.0
         GitCommit:        
-----> docker daemon info: 
       Client:
        Debug Mode: true
       
       Server:
        Containers: 0
         Running: 0
         Paused: 0
         Stopped: 0
        Images: 1
        Server Version: 19.03.8
        Storage Driver: overlay2
         Backing Filesystem: <unknown>
         Supports d_type: true
         Native Overlay Diff: true
        Logging Driver: json-file
        Cgroup Driver: cgroupfs
        Plugins:
         Volume: local
         Network: bridge host ipvlan macvlan null overlay
         Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
        Swarm: inactive
        Runtimes: runc
        Default Runtime: runc
        Init Binary: docker-init
        containerd version: 
        runc version: 
        init version: 
        Security Options:
         apparmor
         seccomp
          Profile: default
        Kernel Version: 5.4.0-29-generic
        Operating System: Ubuntu 20.04 LTS
        OSType: linux
        Architecture: x86_64
        CPUs: 1
        Total Memory: 981.3MiB
        Name: test-deploy
        ID: QKQF:OZTY:2YRA:NULQ:QCVN:AG6I:Q52W:D3EU:56RF:3CNP:CBJS:WESG
        Docker Root Dir: /var/lib/docker
        Debug Mode: false
        Registry: https://index.docker.io/v1/
        Labels:
        Experimental: false
        Insecure Registries:
         127.0.0.0/8
        Live Restore Enabled: false
       
WARNING: No swap limit support
-----> sigil version: 0.6.0
-----> herokuish version: 
       herokuish: 0.5.14
       buildpacks:
         heroku-buildpack-multi     v1.0.0
         heroku-buildpack-ruby      v215
         heroku-buildpack-nodejs    v171
         heroku-buildpack-clojure   v84
         heroku-buildpack-python    v170
         heroku-buildpack-java      v66
         heroku-buildpack-gradle    v31
         heroku-buildpack-scala     v87
         heroku-buildpack-play      v26
         heroku-buildpack-php       v176
         heroku-buildpack-go        v142
         buildpack-nginx            v13
-----> dokku version: dokku version 0.20.4
-----> dokku plugins: 
       plugn: 0.5.0
         00_dokku-standard    0.20.4 enabled    dokku core standard plugin
         20_events            0.20.4 enabled    dokku core events logging plugin
         app-json             0.20.4 enabled    dokku core app-json plugin
         apps                 0.20.4 enabled    dokku core apps plugin
         builder-dockerfile   0.20.4 enabled    dokku core builder-dockerfile plugin
         builder-herokuish    0.20.4 enabled    dokku core builder-herokuish plugin
         buildpacks           0.20.4 enabled    dokku core buildpacks plugin
         certs                0.20.4 enabled    dokku core certificate management plugin
         checks               0.20.4 enabled    dokku core checks plugin
         common               0.20.4 enabled    dokku core common plugin
         config               0.20.4 enabled    dokku core config plugin
         docker-options       0.20.4 enabled    dokku core docker-options plugin
         domains              0.20.4 enabled    dokku core domains plugin
         enter                0.20.4 enabled    dokku core enter plugin
         git                  0.20.4 enabled    dokku core git plugin
         logs                 0.20.4 enabled    dokku core logs plugin
         network              0.20.4 enabled    dokku core network plugin
         nginx-vhosts         0.20.4 enabled    dokku core nginx-vhosts plugin
         plugin               0.20.4 enabled    dokku core plugin plugin
         proxy                0.20.4 enabled    dokku core proxy plugin
         ps                   0.20.4 enabled    dokku core ps plugin
         repo                 0.20.4 enabled    dokku core repo plugin
         resource             0.20.4 enabled    dokku core resource plugin
         scheduler-docker-local 0.20.4 enabled    dokku core scheduler-docker-local plugin
         shell                0.20.4 enabled    dokku core shell plugin
         ssh-keys             0.20.4 enabled    dokku core ssh-keys plugin
         storage              0.20.4 enabled    dokku core storage plugin
         tags                 0.20.4 enabled    dokku core tags plugin
         tar                  0.20.4 enabled    dokku core tar plugin
         trace                0.20.4 enabled    dokku core trace plugin
```

rvanlaar commented:
There were some issues with the debconf configuration. It wasn't picked up by cloudinit because the required syntax and place in the yaml was different.

This is now fixed. I tested this on Digital Ocean. 
As a recommendation, add a new doc page just for cloudinit, and provide it as a downloadable file. It would help users for multiple cloud providers

btw, I added [ci-skip] in my git commit message, but the checks are still running.
